compiler/src/iree/compiler/Codegen/Common/GPU/GPUDistributionPatterns.cpp

```diff
@@ -20,6 +22,82 @@ using VectorValue = TypedValue<VectorType>;
 
 namespace {
 
+static std::optional<int64_t> findDimInLayout(LayoutAttr layout,
```

Comment:
Can you add an example to show what this function does?
Also, what do you think about making the name more specific like `findDimShape`?